src/core/compact_object.h

```diff
@@ -445,13 +448,21 @@ class CompactObj {
     };
   } __attribute__((packed));
 
+  struct JsonConsT {
+    JsonType* json_ptr;
+    size_t bytes_used;
+  };
+
+  struct FlatJsonT {
+    uint32_t json_len;
+    uint8_t* flat_ptr;
+  };
+
   struct JsonWrapper {
     union {
-      JsonType* json_ptr;
-      uint8_t* flat_ptr;
+      JsonConsT cons;
+      FlatJsonT flat;
     };
-    uint32_t json_len = 0;
```

Comment:
why not keep both fields common for both encodings?

Reply:
IMO len is how many characters. The other one is how many bytes. I find it a little clearer :shrug: